portal/templates/profile_macros.html

```diff
@@ -462,6 +463,116 @@ <h4 class="text-success registration-label">{{_('registered')}}</h4>
+                          </div>
+                        </div>
+                        {%- endif %}
+                        {% if current_user and (current_user.has_role(ROLE.STAFF) or current_user.has_role(ROLE.ADMIN) or current_user.has_role(ROLE.INTERVENTION_STAFF)) %}
+                         <div class="tab">
+                          <input id="tab-three" name="tabGroup" type="radio" data-validate="false" class="tab-input">
+                          <label id="lbEmailLog" for="tab-three" class="tab-label">{{_("Email audit log")}}</label>
+                          <div class="tab-content">
+                            {{profileEmailLog(person)}}
+                          </div>
+                        </div>
+                        {% endif %}
+                    </div>
+                </div>
+            </div>
+        </div>
+    {%- elif person and person.has_role(ROLE.STAFF) and current_user and current_user.has_role(ROLE.STAFF_ADMIN)-%}
+        <div class="row">
+            <div class="col-md-12 col-xs-12">
+                <div class="profile-item-container">
+                    <h4 id="communicationsLoc" class="profile-item-title index-item">{{ _("Communications") }}</h4>
+                    {%- if person and person.has_role(ROLE.WRITE_ONLY) -%}
+                        <div id="communicationsContainer" class="tabs-container">
+                            <div class="tab">
+                                <input id="tab-four" name="tabStaffGroup" type="radio" data-validate="false" class="tab-input">
+                                <label for="tab-four" class="tab-label">{{_("Send reset password email to user")}}</label>
+                                <div class="tab-content">
+                                    {{resetPassword(person)}}
+                                </div>
+                            </div>
+                            <div class="tab">
+                                <input id="tab-five" name="tabStaffGroup" type="radio" data-validate="false" class="tab-input">
+                                <label id="lbStaffRegEmail" for="tab-five" class="tab-label">{{_("Send registration email to user")}}</label>
+                                <div class="tab-content">
+                                    {{profileStaffRegistrationEmail(person)}}
+                                </div>
+                            </div>
+                        </div>
+                    {%- else -%}
+                        <div class="flex">
+                            <div id="resetPasswordContainer" class="flex-item">
+                                <p class="text-muted communication-prompt">{{_("Send reset password email to staff")}}</p>
+                                {{resetPassword(person)}}
+                            </div>
+                        </div>
+                    {%- endif -%}
+                </div>
+            </div>
+        </div>
+    {%- endif -%}
+    <script>
```

Comment:
Any chance on moving this to a separate file? :)

Reply:
Ivan, it has been on my mind to do this - need to think this over how best to do this as the macros are shared by multiple pages.  I will do a refactoring in a different PR.